network/network-address/src/lib.rs

```diff
@@ -321,6 +321,45 @@ impl NetworkAddress {
             .push(Protocol::Handshake(handshake_version))
     }
 
+    /// Check that a `NetworkAddress` looks like a typical LibraNet address with
+    /// associated protocols.
+    pub fn is_libranet_addr(&self) -> bool {
+        use Protocol::*;
+
+        let suffix = if self.len() == 3 {
```

Comment:
comments would help to avoid reverse engineering code :) (what's suffix, prefix)

also, it looks like the NetworkAddr only has 3 components if it's a test. Can we also add a `&& cfg!(test)` ?